templates/include/yarp/ast.h.erb

```diff
@@ -84,14 +84,15 @@ typedef struct yp_<%= node.human %> {
     yp_node_t base;
 <%- node.params.grep_v(FlagsParam).each do |param| -%>
     <%= case param
-    in NodeParam | OptionalNodeParam then "struct #{param.c_type} *#{param.name}"
-    in NodeListParam then "struct yp_node_list #{param.name}"
-    in LocationListParam then "yp_location_list_t #{param.name}"
-    in ConstantParam then "yp_constant_id_t #{param.name}"
-    in ConstantListParam then "yp_constant_id_list_t #{param.name}"
-    in StringParam then "yp_string_t #{param.name}"
-    in LocationParam | OptionalLocationParam then "yp_location_t #{param.name}"
-    in UInt32Param then "uint32_t #{param.name}"
+    when NodeParam, OptionalNodeParam then "struct #{param.c_type} *#{param.name}"
```

Comment:
It looks like you're removing all pattern matching and single-line method definitions. I'm assuming this is to get CI running for JRuby/TruffleRuby. In this case can you document what Ruby features/version we can use for internal development?

Reply:
@kddnewton until TR implements YARP they have almost no support for pattern matching.  JRuby supports it but we need both to work with the same code.

Reply:
Yeah for sure, makes sense. I just wasn't sure if it was just pattern-matching and single-line methods or if it was anything else as well.

Reply:
Does JRuby or TruffleRuby not have support for single-line methods?

Reply:
> It looks like you're removing all pattern matching and single-line method definitions. I'm assuming this is to get CI running for JRuby/TruffleRuby.

Correct.

> In this case can you document what Ruby features/version we can use for internal development?

Where should I document that? Not that the CI will catch such issues and of course a newer release of jruby/truffleruby might support some of that.

> Does JRuby or TruffleRuby not have support for single-line methods?

No for TruffleRuby, we're counting on adopting YARP to solve it: https://github.com/oracle/truffleruby/issues/3038

Reply:
JRuby supports all 3.1 syntax features sans fixing argument order execution and flip-flops.  The argument order execution I do not think is going to be a source of problems for years but we will get to it.  flip-flops no one uses (at least in libraries).

Reply:
`CONTRIBUTING.md` would be a good place.